contractsV5/KyberHintHandler.sol

```diff
@@ -0,0 +1,352 @@
+               bpsSoFar += splitValuesBps[i];
+               indexToContinueFrom += 2;
+           }
+       }
+
+       require((bpsSoFar == BPS) || (bpsSoFar == 0), "BPS > 10000");
+
+       reservesHint.addresses = addresses;
+       reservesHint.splitValuesBps = splitValuesBps;
+
+       return indexToContinueFrom;
+    }
+
+    function encodeOpcode(TradeType tradeType) internal pure returns (bytes memory) {
+        if (tradeType == TradeType.MaskIn) {
+            return MASK_IN_OPCODE;
+        } else if (tradeType == TradeType.MaskOut) {
+            return MASK_OUT_OPCODE;
+        } else if (tradeType == TradeType.Split) {
+            return SPLIT_TRADE_OPCODE;
+        }
+    }
+
+    function parseEthToTokenHint(bytes calldata hint)
+        external
+        view
+        returns(
+            TradeType tradeType,
+            bytes8[] memory reserveIds,
+            uint[] memory splits
+        )
+    {
+        IKyberReserve[] memory addresses;
+
+        (tradeType, addresses, splits) = parseHintE2T(hint);
+
+        reserveIds = new bytes8[](addresses.length);
+
+        for (uint i = 0; i < addresses.length; i++) {
+            reserveIds[i] = convertAddressToReserveId(address(addresses[i]));
+        }
+    }
+        
+    function parseTokenToEthHint(bytes calldata hint)
+        external
+        view
+        returns(
+            TradeType tradeType,
+            bytes8[] memory reserveIds,
+            uint[] memory splits
+        )
+    {
+        IKyberReserve[] memory addresses;
+
+        (tradeType, addresses, splits) = parseHintT2E(hint);
+
+        reserveIds = new bytes8[](addresses.length);
+
+        for (uint i = 0; i < addresses.length; i++) {
+            reserveIds[i] = convertAddressToReserveId(address(addresses[i]));
+        }
+    }
+
+    function parseTokenToTokenHint(bytes calldata hint)
+        external
+        view
+        returns(
+            TradeType t2eType,
+            bytes8[] memory t2eReserveIds,
+            uint[] memory t2eSplits,
+            TradeType e2tType,
+            bytes8[] memory e2tReserveIds,
+            uint[] memory e2tSplits
+        )
+    {
+        IKyberReserve[] memory t2eAddresses;
+        IKyberReserve[] memory e2tAddresses;
+
+        (
+            t2eType,
+            t2eAddresses,
+            t2eSplits,
+            e2tType,
+            e2tAddresses,
+            e2tSplits
+        ) = parseHintT2T(hint);
+
+        t2eReserveIds = new bytes8[](t2eAddresses.length);
+        e2tReserveIds = new bytes8[](e2tAddresses.length);
+
+        for (uint i = 0; i < t2eAddresses.length; i++) {
+            t2eReserveIds[i] = convertAddressToReserveId(address(t2eAddresses[i]));
+        }
+        for (uint i = 0; i < e2tAddresses.length; i++) {
+            e2tReserveIds[i] = convertAddressToReserveId(address(e2tAddresses[i]));
+        }
+    }
+
+    function convertReserveIdToAddress(bytes8 reserveId) internal view returns (address) {}
```

Comment:
this should have empty body.
so this contract can't be deployed without implementing the functions.

Reply:
Fixed